### ISSUE TITLE
If you have C/C++ from GCC Adding Fortran should be easy

### ISSUE THREAD

**Issue by zbeekman** [judge0/ide#15]
Right? If so consider this a feature request 😛

**Comment by hermanzdosilovic** [comment]
Sure! :smiley: 

I suppose [here](https://github.com/judge0/api-base/blob/master/Dockerfile#L30) Fortran should be listed along with c and c++?

How do I compile and run Fortran with gcc? Can you also give me an example with hello world program?

Is there any alternative to gcc for Fortran?

**Comment by zbeekman** [comment]
> I suppose here Fortran should be listed along with C and C++?

Exactly, the configure flag needs to be changed to `--enable-languages=c,c++,fortran`.

> How do I compile and run Fortran with GCC? Can you also give me an example with hello world program?

Many flags are shared between gcc and gfortran. Here is a simple hello world program.

```fortran
program main
!! nothing special about naming it `main` but might comfort errant C programmers
  use ISO_FORTRAN_ENV, only: stdout => output_unit, stderr => error_unit ! Get stdout and std err file handles and rename them to stdout and stderr
  implicit none ! Enforce strong typing, no implicit typing

  write(stdout, '(A)') "Hello world!"
  write(stderr, '(A, I0, A, I0, A)') "GCC doesn't ship with OpenCoarrays, so we can only only have a single Coarray Fortran image. Hello from image ", this_image(), " of ", num_images(), "."
  write(stderr, '(A)') "You can install OpenCoarrays and then compile with `-fcoarray=lib` and link the compiled executables to the opencoarrays library for multi image support"
end program
```

You can compile with `gfortran -o foo foo.f90 -std=f2008 -ffree-line-length-none -fcoarray=single`.  The `-stdf2008` flag isn't really needed, and if you get rid of the calls to `this_image()` and `num_images()` you don't need to pass `-fcoarray=single`. If you shorten the lines below 132--I think--characters then you don't need to pass `-ffree-line-length-none`.

> Is there any alternative to GCC for Fortran?

What do you mean? Are you referring to other compiler vendors? Or how you invoke the GCC Fortran compiler? You invoke via `gfortran`. As far as other compiler vendors go, yeah, there are a whole bunch. These days the main ones are:

 - GCC
 - Intel
 - Cray
 - IBM
 - NAG (numerical algorithms group)
 - PGI/NVidia (Portland group was bought by NVidia)
 
There used to be another (fork of GFortran?) G95, but I don't think it's maintained anymore. There's also [FLANG](https://github.com/flang-compiler/flang) with a LLVM backend, maintained by PGI, but it has a lot of catching up to do standards-conformance-wise, even more than PGIs commercial compiler. Notably, Intel offers a non-commercial free license for their Fortran compiler on Linux systems.

**Comment by hermanzdosilovic** [comment]
Thank you! I will add Fortran in the next release :smiley:
